commands/update/nodepool/command.go

```diff
@@ -194,29 +175,40 @@ func updateNodePool(args Arguments) (*result, error) {
 		return nil, microerror.Mask(err)
 	}
 
-	requestBody := &models.V5ModifyNodePoolRequest{}
-	if args.Name != "" {
-		requestBody.Name = args.Name
-	}
-	if args.ScalingMin != 0 || args.ScalingMax != 0 {
-		requestBody.Scaling = &models.V5ModifyNodePoolRequestScaling{}
-	}
-	if args.ScalingMin != 0 {
-		requestBody.Scaling.Min = args.ScalingMin
-	}
-	if args.ScalingMax != 0 {
-		requestBody.Scaling.Max = args.ScalingMax
-	}
-
 	clusterID, err := clustercache.GetID(args.APIEndpoint, args.ClusterNameOrID, clientWrapper)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
 	auxParams := clientWrapper.DefaultAuxiliaryParams()
 	auxParams.ActivityName = activityName
+	existingNP, err := clientWrapper.GetNodePool(clusterID, args.NodePoolID, auxParams)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
 
-	response, err := clientWrapper.ModifyNodePool(clusterID, args.NodePoolID, requestBody, auxParams)
+	updated := false
+	modifyRequestBody := &models.V5ModifyNodePoolRequest{}
+	{
+		if args.Name != "" && args.Name != existingNP.Payload.Name {
+			modifyRequestBody.Name = args.Name
+			updated = true
+		}
+		if args.ScalingMin != 0 && args.ScalingMin != existingNP.Payload.Scaling.Min ||
```

Comment:
is the name and the scaling values the only things that can be updated? What about vmsize?

Reply:
Currently, that can't be updated
https://docs.giantswarm.io/api/#operation/modifyNodePool